### DISCUSSION TITLE
Mouse enter

### DISCUSSION THREAD

**Discussion by ddkwork** [cogentcore/core#739] [General]
Hi, how to set font size when mouse enter and leave

```
o := gi.NewButton(tb).SetText("open").SetIcon(icons.FolderOpen)
			o.On(events.MouseEnter, func(e events.Event) {
				println("MouseEnter")
				o.Styles.Font.Face.Size += 3 
				o.ApplyStylePrefs()
			})
			o.On(events.MouseLeave, func(e events.Event) {
				println("MouseLeave")
				o.Styles.Font.Face.Size -= 3
				o.ApplyStylePrefs()
			})
```

![screenshots](https://github.com/goki/gi/assets/19886504/d2712f94-86a1-4926-b442-19135c342014)

Changing the font size and color through the entry and exit events of the mouse can animate controls such as buttons

**Comment by kkoreilly** [comment]
A better approach to this would be something like this, but there is currently a bug that prevents this from working. I will fix this soon, so do not try to figure out why this code is not working now; it should work soon.

```go
	o := gi.NewButton(inputs).SetText("open").SetIcon(icons.FolderOpen)
	o.OnWidgetAdded(func(w gi.Widget) {
		if lb, ok := w.(*gi.Label); ok {
			lb.Style(func(s *styles.Style) {
				if o.StateIs(states.Hovered) {
					s.Font.Size = units.Dp(17)
				} else {
					s.Font.Size = units.Dp(14)
				}
			})
		}
	})
```

**Comment by ddkwork** [comment]
> A better approach to this would be something like this, but there is currently a bug that prevents this from working. I will fix this soon, so do not try to figure out why this code is not working now; it should work soon.
> 
> ```go
> 	o := gi.NewButton(inputs).SetText("open").SetIcon(icons.FolderOpen)
> 	o.OnWidgetAdded(func(w gi.Widget) {
> 		if lb, ok := w.(*gi.Label); ok {
> 			lb.Style(func(s *styles.Style) {
> 				if o.StateIs(states.Hovered) {
> 					s.Font.Size = units.Dp(17)
> 				} else {
> 					s.Font.Size = units.Dp(14)
> 				}
> 			})
> 		}
> 	})
> ```

In addition to implementing button animations, this feature should be useful for implementing such widgets https://primevue.org/dock/
